source/proof_address/lib/proof_address.rb

```diff
@@ -44,17 +48,40 @@ def post_callback(callback_body:)
         Faraday.post(
           callback_url,
           callback_body.to_json,
-          "X-API-AUTH-TOKEN" => ENV.fetch("IDP_API_AUTH_TOKEN"),
+          "X-API-AUTH-TOKEN" => api_auth_token,
           "Content-Type" => 'application/json',
           "Accept" => 'application/json'
         )
       end
     end
 
+    def api_auth_token
+      @api_auth_token ||= ENV.fetch("IDP_API_AUTH_TOKEN") do
+        ssm_helper.load('address_proof_result_token')
+      end
```

Comment:
the block form here is so we can defer this and *not* call `ssm_helper.load` unless we need to